magma/lib/magma/server/query.rb

```diff
@@ -11,7 +11,7 @@ def action
         return success(Magma::Predicate.to_json, "application/json")
       end
       question = Magma::Question.new(
-        @project_name, @params[:query],
+        @project_name, @params[:query].dup,
```

Comment:
Will this only be a shallow copy and so vulnerable to internal arrays being shifted?